pkg/workload/admissionchecks.go

```diff
@@ -87,27 +87,27 @@ func FindAdmissionCheck(checks []kueue.AdmissionCheckState, checkName string) *k
 }
 
 // ResetChecksOnEviction sets all AdmissionChecks to Pending
-func ResetChecksOnEviction(w *kueue.Workload) {
+func ResetChecksOnEviction(w *kueue.Workload, now time.Time) {
 	checks := w.Status.AdmissionChecks
 	for i := range checks {
 		checks[i] = kueue.AdmissionCheckState{
 			Name:               checks[i].Name,
 			State:              kueue.CheckStatePending,
-			LastTransitionTime: metav1.NewTime(time.Now()),
+			LastTransitionTime: metav1.NewTime(now),
 			Message:            "Reset to Pending after eviction. Previously: " + string(checks[i].State),
 		}
 	}
 }
 
 // SetAdmissionCheckState - adds or updates newCheck in the provided checks list.
-func SetAdmissionCheckState(checks *[]kueue.AdmissionCheckState, newCheck kueue.AdmissionCheckState) {
+func SetAdmissionCheckState(checks *[]kueue.AdmissionCheckState, newCheck kueue.AdmissionCheckState, now time.Time) {
```

Comment:
Hmm. Maybe these changes aren't necessary We could propagate time in `newCheck kueue.AdmissionCheckState`

https://github.com/kubernetes-sigs/kueue/blob/0c9889edbd3f25b115db4268ac8d646d5a58aad2/pkg/workload/admissionchecks.go#L109-L111

WDYT?

Reply:
Please check this case if it's really need.